src/com/lld/inmemorycache/model/impl/Cache.java

```diff
@@ -0,0 +1,56 @@
+package com.lld.inmemorycache.model.impl;
+
+import com.lld.inmemorycache.model.AbstractCache;
+import com.lld.inmemorycache.model.IEvictionPolicy;
+import com.lld.inmemorycache.model.IStorage;
+
+public class Cache extends AbstractCache {
+
+    public Cache(IStorage storage, IEvictionPolicy evictionPolicy, int capacity)
+    {
+        this.storage = storage;
+        this.evictionPolicy = evictionPolicy;
+        this.capacity = capacity;
+    }
+    @Override
+    public boolean put(String key, String value) {

```

Comment:
The operations across storage and eviction layer are not synchronized. A lock at the cache level has to guard these operations together.